### PR TITLE
Fix Gapless playback for SB1

### PR DESCRIPTION
**Pull request by michaldie**
Fix to address this issue : 

https://github.com/Logitech/slimserver/issues/366

the code does work as described in the issue - 
however it is not very pretty - but then I am no Perl programmer ..
comments welcome

Michael

### CODE REVIEW

File: Slim/Player/Squeezebox1.pm [hunk 1]
```diff
@@ -142,17 +142,17 @@ sub isReadyToStream {
...
-					&& $playingSong->samplesize == $song->samplesize
-					&& $playingSong->samplerate == $song->samplerate
-				) {
-					return 1;
-				}
-			}
-		}
-	}
-	
+                        if ( $playingSong ) {
```
Comment (michaelherger) [inline Slim/Player/Squeezebox1.pm]: Could you please revert whitespace changes? Make sure you're using tabs rather than spaces.

Comment (michaelherger) [reply Slim/Player/Squeezebox1.pm]: You should probably get the two `currentTrack` objects beforehand and check their existence, too.

```perl
my $playingTrack = $playingSong->currentTrack;
my $songTrack = $song->currentTrack;

if ($playingTrack && $songTrack && $playingTrack->channels == $songTrack->channels && ...
```

File: Slim/Player/Squeezebox1.pm [hunk 2]
```diff
@@ -142,17 +142,17 @@ sub isReadyToStream {
...
+                                if (   $playingSong->currentTrack->channels   == $song->currentTrack->channels
+                                        && $playingSong->currentTrack->samplesize == $song->currentTrack->samplesize
+                                        && $playingSong->currentTrack->samplerate == $song->currentTrack->samplerate
+                                ) {
+                                        return 1;
+                                }
+                        }
+                }
+        }
+        main::DEBUGLOG && logger('player.playlist')->debug("Restart decoder required");
```
Comment (michaelherger) [inline Slim/Player/Squeezebox1.pm]: I'd say the `player` logging object would be correct here.

**Review by michaelherger — CHANGES_REQUESTED** [review]
Please apply the minor changes commented inline. The patch itself overall makes sense. Thanks!

**Comment by michaldie** [comment]
Ok, new commit is in place.

**Review by mherger — APPROVED** [review]

**Comment by mherger** [comment]
Thanks!

**Pull request merged**